examples/aptos-friend/frontend/components/TradeShare.tsx

```diff
@@ -0,0 +1,104 @@
+import { useWallet } from "@aptos-labs/wallet-adapter-react";
+import { useState } from "react";
+import { useWalletClient } from "@thalalabs/surf/hooks";
+
+import { Button } from "@/components/ui/button";
+import { ABI } from "@/utils/abi";
+import { Card, CardHeader, CardTitle, CardContent } from "@/components/ui/card";
+import { Toggle } from "@/components/ui/toggle";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { useGetHolding, useGetIssuerObjectAddress } from "@/hooks/useHolding";
+import { aptosClient } from "@/utils/aptosClient";
+import { useToast } from "@/components/ui/use-toast";
+
+type TradeShareProps = {
+  issuerAddress: `0x${string}`;
+};
+
+export function TradeShare({ issuerAddress }: TradeShareProps) {
+  const { client: walletClient } = useWalletClient();
+  const { account } = useWallet();
+  const { toast } = useToast();
+  const holding = useGetHolding(issuerAddress, account?.address as `0x${string}`);
```

Comment:
why not just changing the holder account address type to `string`?

Reply:
surf requires all address to be `0x${string}`